efs.tf

```diff
@@ -13,7 +14,7 @@ resource "aws_efs_file_system" "efsEncryptedFalse" {
     Name = "encrypted"
   }
```

Comment:
**Accurics remediation**
 
**[HIGH]: Ensure that your Amazon EFS file systems are encrypted**
 
**Vulnerability**
Enable encryption of your EFS file systems in order to protect your data and metadata from breaches or unauthorized access and fulfill compliance requirements for data-at-rest encryption within your organization.
 
**Remediation**
You can enable encryption of data at rest when creating an Amazon EFS file system. Note: There is no functionality that allows you to encrypt existing EFS if the encryption wasn't enabled during the creation of Amazon EFS process. In order to encrypt an existing Amazon EFS, you need to create a new Amazon EFS and copy all the data from the existing Amazon EFS onto the new one with encryption enabled. Encrypting a File System at Rest Using the Console. You can choose to enable encryption at rest for a file system when you create it. The following procedure describes how to enable encryption for a new file system when you create it from the console. To encrypt a new file system on the console: 1. Open the Amazon Elastic File System console at https://console.aws.amazon.com/efs/. 2. Choose Create file system to open the file system creation wizard. 2a. For Step 1: Configure file system access, choose your VPC, create your mount targets, and then choose Next Step. 2b. For Step 2: Configure optional settings, add any tags, choose your performance mode, check the box to encrypt your file system, and then choose Next Step. 2c. For Step 3: Review and create, review your settings, and choose Create File System. 3. Check Enable encryption checkbox and choose the name of the AWS Key from Select KMS master key dropdown list to enable encryption using your own KMS CMK key.

source: accurics-onprem